RFCS: update readme

petermattis opened the pull request:
Add initial sections that go into more detail of the when and how of
RFCs and mention the benefits of prototyping. Add a separate
PROTOTYPING.md doc that goes into more details.

Trim the "RFC Process" section, replacing the FCP with the judgment of
the author and reviewers. Trim the "RFC Status" section as that level of
formalism seems like overkill. The template lists the states and
mistakes here aren't really a problem.

Release note: none

petermattis committed:
RFCS: update readme

Add initial sections that go into more detail of the when and how of
RFCs and mention the benefits of prototyping. Add a separate
PROTOTYPING.md doc that goes into more details.

Trim the "RFC Process" section, replacing the FCP with the judgment of
the author and reviewers. Trim the "RFC Status" section as that level of
formalism seems like overkill. The template lists the states and
mistakes here aren't really a problem.

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21781)
<!-- Reviewable:end -->

petermattis commented:
This took longer to get out that I anticipated. Hopefully it is in line with expectations from offline discussions.

vivekmenezes commented on docs/RFCS/PROTOTYPING.md:
```diff
@@ -0,0 +1,68 @@
+you're trying to learn. Comments? Only if you, the prototype author,
+need them. The real focus of the prototype is exploring the unseen
+corners of the problem space in order to reveal where the dragons are
+lurking.
```
The Goal here might be to understand the hardest problems that are blocking your work.

vivekmenezes commented on docs/RFCS/PROTOTYPING.md:
```diff
@@ -0,0 +1,68 @@
+overcome yourself. The benefit of the prototype truly shines when this
+happens and you'll have a deep understanding of the problem which you
+can use to have a focused discussion with other engineers. Share your
+failed prototypes. The failure implies learning and that learning
```
I think it is worth stressing here that it is encouraged to fail fast and we want you to be open about sharing failure. You as an engineer can move faster by soliciting for help here.

vivekmenezes commented on docs/RFCS/README.md:
```diff
@@ -1,209 +1,94 @@
+often requires multiple engineers worth of effort). Note that this
+guidance is intentionally vague. Many complex projects spread over
+multiple PRs do not require an RFC, though they do require adequate
+communication and discussion.
```
When there are a number of alternatives where implementing each alternative could take a lot of time it is worth discussing it in an RFC

vivekmenezes commented on docs/RFCS/README.md:
```diff
@@ -1,209 +1,94 @@
+   `completed`. If subsequent developments render an RFC obsolete,
+   change its status to `obsolete`. When you mark a RFC as obsolete,
+   ensure that its text references the other RFCs or PRs that make it
+   obsolete.
```
I think it's worth having a "postponed" status to reflect projects that were postponed due to opportunity cost.

knz commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3hqaLUZv9mU2VGTFwU:b-1ho54o) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
> ```Markdown
> guidance is intentionally vague. Many complex projects spread over
> multiple PRs do not require an RFC, though they do require adequate
> communication and discussion.
> ```

A point loudly made by PM a while ago when we last visited the RFC process was that a RFC is also strongly recommended if the change impacts user experience in any way.

---

*[docs/RFCS/README.md, line 70 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqyLGklzUDMcHt3nt:-L3hqyLHp8ytuRWT2yXw:buqhmz8) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L70)):*
> ```Markdown
> 
> 4. Go through the PR review, iterating on the RFC to answer questions
>    and concerns from the reviewer(s). The duration of this process
> ```

The notion of a shepherd was lost in translation. I'm ok with a simpler process but having one person responsible for moving the discussion along and having the option to keep this person separate from the author(s) is still very much needed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hk2zGIbJztZ6jyr8_-r1-36:-L3hsosJXW8-VFjVLNpU:b-oprs3p) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

When there are a number of alternatives where implementing each alternative could take a lot of time it is worth discussing it in an RFC
</blockquote></details>

I thought about this a lot while writing this section. Discussing numerous alternatives is exactly when prototyping can help sort out which are viable and which are not. 

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3ht4sUoVyojqfPNhg5:b-i9eaou) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

A point loudly made by PM a while ago when we last visited the RFC process was that a RFC is also strongly recommended if the change impacts user experience in any way.
</blockquote></details>

How do we define "impacts user experience" in a way that doesn't include pretty much every change?

---

*[docs/RFCS/README.md, line 70 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqyLGklzUDMcHt3nt:-L3htbn97l_7qwpCIk1H:buclo2d) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L70)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

The notion of a shepherd was lost in translation. I'm ok with a simpler process but having one person responsible for moving the discussion along and having the option to keep this person separate from the author(s) is still very much needed.
</blockquote></details>

Hmm, ok. Let me take a look.

---

*[docs/RFCS/README.md, line 94 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hk2zGIbJztZ6jyr8_-r1-94:-L3htN75L7s_V1Y5mAn9:b-tfa4gw) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L94)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

I think it's worth having a "postponed" status to reflect projects that were postponed due to opportunity cost.
</blockquote></details>

I didn't remove any of the states, just removed them from this doc. Do you think it is necessary to be so precisely descriptive about what each state means?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/PROTOTYPING.md, line 54 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqgY1aTqCWUsYVtnx:-L3hqgY1aTqCWUsYVtny:bs9i3pu) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/PROTOTYPING.md#L54)):*
> ```Markdown
> engineers that an idea has been explored.
> 
> An RFC is a high-level description. There is utility in
> ```

Maybe expand a little bit and mention that the description doesn't need to be formally accurate, as long as it intuitively makes sense. In the past there were some RFCs which went way too far in defining things formally and I just got lost in the descriptions - it was hard to distinguish the "meat" of the document from the rest.

Conversely, I would add that some details are worth specifying if they involve choices that are likely to lead to bike-shedding (for example, new SQL syntax). It's better if that discussion is had over the RFC.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-:-L3ht54SRP0jzkQkzvSR:b-uwkowd)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3huJANMjH6CecidV96:b-g11zmp) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

How do we define "impacts user experience" in a way that doesn't include pretty much every change?
</blockquote></details>

Was the desire for an RFC for changes which "impact user experience" due to the desire to have discussion? I don't want to do away the discussion. I also don't want to mandate that RFCs are the only forum for such discussions. That feels too rigid and proscriptive. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3huEAKP3viXducS41e:bg2wgkb) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Was the desire for an RFC for changes which "impact user experience" due to the desire to have discussion? I don't want to do away the discussion. I also don't want to mandate that RFCs are the only forum for such discussions. That feels too rigid and proscriptive. 
</blockquote></details>

Good question. You have multiple elements for an answer to this quesiton here:
- the draft in #19175.
- https://www.cockroachlabs.com/docs/stable/contribute-to-cockroachdb.html  in particular this part:

  "A project is likely high complexity (and thus needs a RFC) if any of the following conditions apply:

  - More than a few users/apps will need to change the way they use CockroachDB as a result.
  - More than a few other CockroachDB contributors will need to change the way they work as a result.
  - The architecture of the software or the project changes in a way visible to more than a handful of developers.
  - The change impacts the cost model of CockroachDB (how much time and how much memory is used for common operations) in a way that's noticeable by users.
  - New software dependencies are introduced.
  - New subdirectories/packages will be created in the repository.
  - You can predict in advance that completing this change will take more than a month."

- the semi-formal definition of functional vs extra-functional behavior in our "Postgres compatibility strategy document" by Andy. I'd argue that any functional behavior change mandates a RFC, not so much for extra-functional changes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3hvuL3sRbiKXRyD0lc:bdwvl4o) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Good question. You have multiple elements for an answer to this quesiton here:
- the draft in #19175.
- https://www.cockroachlabs.com/docs/stable/contribute-to-cockroachdb.html  in particular this part:

  "A project is likely high complexity (and thus needs a RFC) if any of the following conditions apply:

  - More than a few users/apps will need to change the way they use CockroachDB as a result.
  - More than a few other CockroachDB contributors will need to change the way they work as a result.
  - The architecture of the software or the project changes in a way visible to more than a handful of developers.
  - The change impacts the cost model of CockroachDB (how much time and how much memory is used for common operations) in a way that's noticeable by users.
  - New software dependencies are introduced.
  - New subdirectories/packages will be created in the repository.
  - You can predict in advance that completing this change will take more than a month."

- the semi-formal definition of functional vs extra-functional behavior in our "Postgres compatibility strategy document" by Andy. I'd argue that any functional behavior change mandates a RFC, not so much for extra-functional changes.

</blockquote></details>

So the desire for an RFC is to ensure that changes to user experience gather evaluators from a wider circle than an engineer's local in-group who would otherwise be sufficient to review the technical aspects.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

petermattis commented:
TFTR

---

Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---

*[docs/RFCS/PROTOTYPING.md, line 34 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hk2zGIbJztZ6jyr8Z-r1-34:-L3i__CBJ4qT5cOMuDUc:bb7jh45) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/PROTOTYPING.md#L34)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

The Goal here might be to understand the hardest problems that are blocking your work. 
</blockquote></details>

Added an additional sentence with slightly different wording.

---

*[docs/RFCS/PROTOTYPING.md, line 48 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hk2zGIbJztZ6jyr8Z-r1-48:-L3iZmngliKaPuYUcq_V:b-cikqju) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/PROTOTYPING.md#L48)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

I think it is worth stressing here that it is encouraged to fail fast and we want you to be open about sharing failure. You as an engineer can move faster by soliciting for help here.
</blockquote></details>

I'm struggling for specific changes to make here. The current text seems to contain the same sentiment. 

---

*[docs/RFCS/PROTOTYPING.md, line 54 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqgY1aTqCWUsYVtnx:-L3iXve9J3gmSVp90i6Q:b-5l12p9) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/PROTOTYPING.md#L54)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Maybe expand a little bit and mention that the description doesn't need to be formally accurate, as long as it intuitively makes sense. In the past there were some RFCs which went way too far in defining things formally and I just got lost in the descriptions - it was hard to distinguish the "meat" of the document from the rest.

Conversely, I would add that some details are worth specifying if they involve choices that are likely to lead to bike-shedding (for example, new SQL syntax). It's better if that discussion is had over the RFC.
</blockquote></details>

Added a brief phrase about not needing formality. I've put a note in `README.md` that SQL syntax deserves discussion in the RFC.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3i_dbtcCv94mFZvdQp:b-xn2xf6) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
Is the admin UI part of user experience? Using the RFC process for changes to the admin UI seems absolutely inappropriate. Is a significant performance improvement part of the user experience?

> A project is likely high complexity (and thus needs a RFC) if any of the following conditions apply:

That list of conditions is exactly what I want to move away from. It feels like mandatory sentencing requirements in the legal arena which are generally panned as being overly restrictive to judges.

---

*[docs/RFCS/README.md, line 70 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqyLGklzUDMcHt3nt:-L3iarUI3Fy4_-vZGiVz:b-stp2gt) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L70)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Hmm, ok. Let me take a look.
</blockquote></details>

I've added a small addition to the sentence above about "partnering with a more experienced engineer" in order to help "shepherd your RFC through the process".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-:-L3iaNSbrIFhOZsrlTFU:bb73zsl)*
<!-- Sent from Reviewable.io -->

couchand commented:
Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3iudThd7CZ_cARwCkS:bhj1j3e) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
> Is the admin UI part of user experience? Using the RFC process for changes to the admin UI seems absolutely inappropriate.

How about "likely inappropriate"?  I think that, as with any other part of the system, it depends.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3iwbd-9WZV48sz7698:b-43w600) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
> That list of conditions is exactly what I want to move away from. It feels like mandatory sentencing requirements in the legal arena which are generally panned as being overly restrictive to judges.

Please take this up with PM. I was just the messenger here. The worry (legitimate in my view) is that engineers with a narrow focus will not necessarily perceive the extent of the UX changes of something that seems local / innocuous to them. There were previous instances where this happened -- again ask PM for examples, either Andy or Diana.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3jKSp3JZ6i5ntTeV_j:b-s5lf3n) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

> That list of conditions is exactly what I want to move away from. It feels like mandatory sentencing requirements in the legal arena which are generally panned as being overly restrictive to judges.

Please take this up with PM. I was just the messenger here. The worry (legitimate in my view) is that engineers with a narrow focus will not necessarily perceive the extent of the UX changes of something that seems local / innocuous to them. There were previous instances where this happened -- again ask PM for examples, either Andy or Diana.
</blockquote></details>

@couchand Yes, "absolutely" was extreme.

@knz Yes, I will discuss with Andy and Diana. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

petermattis commented:
I believe I've addressed all of the comments. PTAL.

---

Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3nJdjSwde13ThLFuG8:bdzbc68) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

@couchand Yes, "absolutely" was extreme.

@knz Yes, I will discuss with Andy and Diana. 
</blockquote></details>

I've conferred with PMs and they seem ok with this wording. Their concern is they want visibility and early notification of user impacting changes. They have no requirement that all such changes be discussed in an RFC. The attendance of PMs at eng meetings mostly addresses their concern.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-:-L3nKaWZWqbl6tMPyBBO:b-cr6cbs)*
<!-- Sent from Reviewable.io -->

a-robinson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-:-L3oIW0SkzVZhhjRUUmr:bnfp4nl)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-:-L3oJEYxCgKhBEKPNe4Y:b-uwkowd)*
<!-- Sent from Reviewable.io -->

dianasaur323 commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3oKXycsgvup_MCOQ_p:bx2d6wl) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I've conferred with PMs and they seem ok with this wording. Their concern is they want visibility and early notification of user impacting changes. They have no requirement that all such changes be discussed in an RFC. The attendance of PMs at eng meetings mostly addresses their concern.
</blockquote></details>

Hi! I was the one who made the assertion, so I'd like to clarify what I meant. Basically, my concern was that the RFC process doesn't necessarily account for how a given change might impact the entire user experience across the entire product. So, it's not that a user experience change should dictate what becomes an RFC, but more that the RFC process should take into account how a change will impact user experience holistically.

To be more specific, an example of this would be discussing how to surface debuggability and visibility in the admin UI and CLI. Just a discussion that this might be necessary is good enough, because then it brings it to the attention of other teams who may have to schedule additional work in order to create a holistic user experience that makes sense. I think we did fine with geo-partitioning, so that was just an example. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->

knz reviewed (approved):
thanks for (re)writing this

petermattis commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[docs/RFCS/README.md, line 36 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781#-L3hqaLRym8uarL_zwry:-L3pGlLjQqJrjtNhSj4y:bl19c5g) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0f76b519f1ddf68e581947fd3fdfaefae0d513b/docs/RFCS/README.md#L36)):*
<details><summary><i>Previously, dianasaur323 (Diana Hsieh) wrote…</i></summary><blockquote>

Hi! I was the one who made the assertion, so I'd like to clarify what I meant. Basically, my concern was that the RFC process doesn't necessarily account for how a given change might impact the entire user experience across the entire product. So, it's not that a user experience change should dictate what becomes an RFC, but more that the RFC process should take into account how a change will impact user experience holistically.

To be more specific, an example of this would be discussing how to surface debuggability and visibility in the admin UI and CLI. Just a discussion that this might be necessary is good enough, because then it brings it to the attention of other teams who may have to schedule additional work in order to create a holistic user experience that makes sense. I think we did fine with geo-partitioning, so that was just an example. 
</blockquote></details>

Ack. I think we're in agreement that we need proper discussion and consideration of the user impact of changes, and that such discussion can occur in an RFC, but can also in other forums. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21781)*
<!-- Sent from Reviewable.io -->
